src/Avalonia.Base/Data/Core/Parsers/ExpressionTreeParser.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+
+namespace Avalonia.Data.Core.Parsers
+{
+    class ExpressionTreeParser
```

Comment:
Does this need to be a separate class? Could this and `ExpressionVisitorNodeBuilder` be combined into a single class?

Reply:
I could combine it, but I like having the visitor implementation in its own class. Keeps the messy logic of the visitor separated from the simpler logic in `ExpressionTreeParser`.